arangod/RestHandler/RestIndexHandler.cpp

```diff
@@ -193,13 +197,89 @@ RestStatus RestIndexHandler::getIndexes() {
       return RestStatus::DONE;
     }
 
+    std::stringstream ap;
+    ap << "Plan/Collections/" << _vocbase.name() << "/" << coll->planId()
+       << "/indexes";
+    auto& ac = _vocbase.server().getFeature<ClusterFeature>().agencyCache();
+    auto [plannedIndexes, idx] = ac.get(ap.str());
+
     TRI_ASSERT(indexes.slice().isArray());
     VPackBuilder tmp;
     tmp.openObject();
     tmp.add(StaticStrings::Error, VPackValue(false));
     tmp.add(StaticStrings::Code,
             VPackValue(static_cast<int>(ResponseCode::OK)));
-    tmp.add("indexes", indexes.slice());
+    tmp.add(VPackValue("indexes"));
+    {
+      VPackArrayBuilder guard(&tmp);
+      for (auto const& i : VPackArrayIterator(indexes.slice())) {
+        tmp.add(i);
+      }
+      for (auto const& pi : VPackArrayIterator(plannedIndexes->slice())) {
+        if (pi.hasKey("isBuilding") && pi.get("isBuilding").getBool()) {
```

Comment:
can be simplified to just
```suggestion
        if (pi.hasKey("isBuilding").isTrue()) {
```